File: mini-coupang/member-server/src/main/java/com/deepblog/member/repository/SellerRepository.java
package com.deepblog.member.repository;

import com.deepblog.member.domain.Seller;
import java.util.Optional;
import org.springframework.data.jpa.repository.JpaRepository;

public interface SellerRepository extends JpaRepository<Seller, Long> {

    Optional<Seller> findByAccountId(Long accountId);

    boolean existsByBusinessRegistrationNumber(String businessRegistrationNumber);
}

File: services/seller-service/src/main/kotlin/com/deepblog/seller/storage/SellerRepository.kt
package com.deepblog.seller.storage

import com.deepblog.seller.domain.Seller
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository

/**
 * Spring Data JPA 인터페이스. 메서드 선언만으로 구현체가 런타임에 생성됨.
 *
 * 커스텀 쿼리는 추후 `findByEmail`, `@Query`, QueryDSL 등으로 확장.
 */
@Repository
interface SellerRepository : JpaRepository<Seller, Long> {
    fun findByEmail(email: String): Seller?
}
